### PR TITLE
[DBEX-TREX] 60247 Migrate separation locations from EVSS to LH

### PR DESCRIPTION
**Pull request by tommasina-va**
This PR adds in the migration of the Benefits Reference Data endpoint from EVSS to Lighthouse behind a feature toggle. 

## Summary

- This work is behind a feature toggle (flipper): YES
- Adds BRD to the API Provider factories
- I'm on the Disability Benefits Experience Team 1 (TREX). We are responsible for migrating EVSS endpoints to LH for Form 526EZ. 

## Related issue(s)

- https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/60247

## Testing done

- [x] *New code is covered by unit tests*
- Old behavior defaulted to using EVSS, new behavior will check the feature toggle to determine which API provider to use. 
  - *What is the testing plan for rolling out the feature?* We will have a Canary launch and release plan

## What areas of the site does it impact?
This impacts the separation_locations method in the DisabilityCompensationFormsController. [/v0/disability_compensation_forms_controller.rb](https://github.com/department-of-veterans-affairs/vets-api/compare/60247-brd-separation-locations?expand=1#diff-cdcaa26c5cfce0d1bda78201f27a4c6eb554d5d45ff50d92eec5f393d3d44f6d)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable) -> Future work

### CODE REVIEW

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by mchae-nava — DISMISSED** [review]
LGTM! 

Sidenote: I had a thought regarding whether or not errors could or should be caught at the factory/provider level. On one hand, wouldn't be great for diagnostic monitoring, but on the other, could be a useful thing to aggregate? Curious on your thoughts

**Review by ericboehs — APPROVED** [review]

File: lib/disability_compensation/providers/brd/lighthouse_brd_provider.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'disability_compensation/providers/brd/brd_provider'
+require 'disability_compensation/responses/intake_sites_response'
+require 'lighthouse/benefits_reference_data/service'
```
Comment (freeheeling) [inline lib/disability_compensation/providers/brd/lighthouse_brd_provider.rb:5]: Add a line break between require statements and class definition.

**Review by freeheeling — APPROVED** [review]

**Review by FonzMP — DISMISSED** [review]
:shipit:

**Review by ericboehs — APPROVED** [review]

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**